include/arch/x86/linker.ld

```diff
@@ -105,6 +106,15 @@ SECTIONS
 	} GROUP_LINK_IN(ROMABLE_REGION)
 
 	_image_text_end = .;
+
+#ifdef CONFIG_VERSION_HEADER
```

Comment:
I see the point. For instance bootloader (for zephyr mcuboot) should be as small as possible  so we usually fighting for every byte less in the footprint.

Reply:
> I see the point. For instance bootloader (for zephyr mcuboot) should be as small as possible so we usually fighting for every byte less in the footprint.

I wouldn't expect the bootloader to use an ELF, but a binary, in which this would presumably not exist.

Reply:
FWIW: as implemented above this is going into ROM and will be linked at runtime.